pkg/clusterapi/proxy.go

```diff
@@ -14,6 +14,50 @@ import (
 //go:embed config/http-proxy.conf
 var proxyConfig string
 
+func proxy(cluster v1alpha1.ClusterSpec) bootstrapv1.ProxyConfiguration {
+	return bootstrapv1.ProxyConfiguration{
+		HTTPSProxy: cluster.ProxyConfiguration.HttpsProxy,
+		NoProxy:    noProxyList(cluster),
+	}
+}
+
+// SetProxyConfigInKubeadmControlPlaneForBottlerocket sets up proxy configuration in kubeadmControlPlane for bottlerocket.
+func SetProxyConfigInKubeadmControlPlaneForBottlerocket(kcp *controlplanev1.KubeadmControlPlane, cluster v1alpha1.ClusterSpec) {
```

Comment:
for all this methods I would be in favor of passing the *Cluster instead of the ClusterSpec, for these reasons:
* We never have a ClusterSpec without a Cluster object so it doesn't make it more reusable or easier to read
* ClusterSpec is passed by value, so it copies the whole thing every time you pass it to a method (not the biggest performance hit, but something)
* ClusterSpec and cluster.Spec can be confusing